src/components/cart-list/index.tsx

```diff
@@ -4,7 +4,24 @@ import { List } from '../general/list';
 import { CartContext } from '../../contexts/Cart';
 
 export const CartList: React.FC = () => {
-  const { cart } = useContext(CartContext);
+  const { cart, setCart } = useContext(CartContext);
+  const changeQuantity = (id: string, amount: number, sign: string) => {
+    setCart((currCart) => {
+      return currCart.map((currItem) => {
+        if (currItem.id === id) {
+          if (sign === '+') {
```

Comment:
you don't need a sign, you can just have amount. if you add a negative number to another one, it's going to subtract from it :) 

you can just have amount (+1 for the increment button, -1 for the decrement button) and still do the zero check. in case this function is used in the future with amounts larger than one, it will be good to check instead if the result is bigger than zero. something like:
```
const newQty = currentItem.quantity + amount > 0 ? currentItem.quantity + amount : 0
{ ...currItem, quantity: newQty}
```